### PR TITLE
use given key when constructing encrypting cipher

### PR DESCRIPTION
**Pull request by ellisjoe**

### CODE REVIEW

File: crypto-core/src/test/java/com/palantir/crypto2/io/CryptoStreamFactoryTest.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.charset.StandardCharsets;
+import org.junit.Test;
+
+public final class CryptoStreamFactoryTest {
+
+    @Test
+    public void testEncryptDecryptJce() throws IOException {
+        System.setProperty(CryptoStreamFactory.FORCE_JCE, "true");
```
Comment (schlosna) [inline crypto-core/src/test/java/com/palantir/crypto2/io/CryptoStreamFactoryTest.java]: Probably want to remove this in an @After

Comment (ellisjoe) [reply crypto-core/src/test/java/com/palantir/crypto2/io/CryptoStreamFactoryTest.java]: done

Comment (ellisjoe) [reply crypto-core/src/test/java/com/palantir/crypto2/io/CryptoStreamFactoryTest.java]: just stuck it at the end of the test since it should only apply to this specific test at the moment

File: crypto-core/src/main/java/com/palantir/crypto2/io/CryptoStreamFactory.java [hunk 2]
```diff
@@ -80,8 +81,9 @@ private static OutputStream createApacheEncryptedStream(OutputStream output, Key
         return new CtrCryptoOutputStream(PROPS, output, secretKey.getEncoded(), iv);
     }
 
-    private static OutputStream createDefaultEncryptedStream(OutputStream output, String algorithm) {
-        SeekableCipher cipher = SeekableCipherFactory.getCipher(algorithm);
+    private static OutputStream createDefaultEncryptedStream(OutputStream output, KeyMaterial keyMaterial,
```
Comment (robert3005) [inline crypto-core/src/main/java/com/palantir/crypto2/io/CryptoStreamFactory.java:95]: Nit: would push all arguments to next line

**Review by robert3005 — APPROVED** [review]

File: crypto-core/src/main/java/com/palantir/crypto2/io/CryptoStreamFactory.java [hunk 3]
```diff
@@ -80,8 +81,9 @@ private static OutputStream createApacheEncryptedStream(OutputStream output, Key
...
+            String algorithm) {
+        SeekableCipher cipher = SeekableCipherFactory.getCipher(algorithm, keyMaterial);
```
Comment (uschi2000) [inline crypto-core/src/main/java/com/palantir/crypto2/io/CryptoStreamFactory.java:97]: the SeekableCipherFactory is dangerous. would either remove the method that does generates key-material, or rename it to getCipherWithGeneratedKeyMaterial, or make the key material an `Optional<>` parameter.

File: crypto-core/src/main/java/com/palantir/crypto2/io/CryptoStreamFactory.java [hunk 4]
```diff
@@ -44,7 +45,7 @@ private CryptoStreamFactory() {}
      * cipher {@code algorithm}. When OpenSSL is available an implementation that uses AES-NI will be returned.
      */
     public static SeekableInput decrypt(SeekableInput encryptedInput, KeyMaterial keyMaterial, String algorithm) {
-        if (!algorithm.equals(AES_ALGORITHM)) {
+        if (!algorithm.equals(AES_ALGORITHM) || Boolean.parseBoolean(System.getProperty(FORCE_JCE))) {
```
Comment (uschi2000) [inline crypto-core/src/main/java/com/palantir/crypto2/io/CryptoStreamFactory.java]: this seems like an orthogonal change? prefer 2 PRs, one to fix the bug and one to allow overriding JCE

Comment (uschi2000) [reply crypto-core/src/main/java/com/palantir/crypto2/io/CryptoStreamFactory.java]: oh, hang on. the FORCE_JCE thing is for testing only? that's a bit invasive, IMO.

Comment (uschi2000) [reply crypto-core/src/main/java/com/palantir/crypto2/io/CryptoStreamFactory.java]: oh, hang on. the FORCE_JCE thing is for testing only? that's a bit invasive, IMO.

**Review by uschi2000 — APPROVED** [review]
